amxmodx/amxmodx.cpp

```diff
@@ -1438,44 +1438,75 @@ static cell AMX_NATIVE_CALL get_plugin(AMX *amx, cell *params) /* 11 param */
 	return -1;
 }
 
-static cell AMX_NATIVE_CALL amx_md5(AMX *amx, cell *params)
+static cell AMX_NATIVE_CALL amx_md5(AMX* amx, cell* params)
 {
-	int len = 0;
-	char *str = get_amxstring(amx, params[1], 0, len);
-	char buffer[33];
+	int strLen;
+	char* pStr = get_amxstring(amx, params[1], 0, strLen);
 
-	MD5 md5;
-	md5.update((unsigned char *)str, len);
-	md5.finalize();
-	md5.hex_digest(buffer);
+	const char* pHash = hashString((const char *)&pStr[0], strLen, Hash_Md5);
 
-	return set_amxstring(amx, params[2], buffer, 32);
-}
+	return set_amxstring(amx, params[2], pHash, 32);
+};
 
-static cell AMX_NATIVE_CALL amx_md5_file(AMX *amx, cell *params)
+static cell AMX_NATIVE_CALL amx_md5_file(AMX* amx, cell* params)
 {
-	int len = 0;
-	char *str = get_amxstring(amx, params[1], 0, len);
-	char buffer[33];
-	char file[255];
+	int fileNameLen;
+	char* pFileName = get_amxstring(amx, params[1], 0, fileNameLen);
 
-	build_pathname_r(file, sizeof(file)-1, "%s", str);
+	char fileLocation[256];
+	build_pathname_r(fileLocation, sizeof(fileLocation) - 1, "%s", pFileName);
 
-	FILE *fp = fopen(file, "rb");
-	
-	if (!fp)
+	const char* pHash = hashFile((const char *)&fileLocation[0], Hash_Md5);
+	if (!pHash)
 	{
-		LogError(amx, AMX_ERR_NATIVE, "Cant open file \"%s\"", file);
+		LogError(amx, AMX_ERR_NATIVE, "There was a problem hashing file (\"%s\"). " \
+			"Perhaps it doesn't exist.", &pFileName[0]);
+
 		return 0;
-	}
+	};
 
-	MD5 md5;
-	md5.update(fp);			//closes for you
-	md5.finalize();
-	md5.hex_digest(buffer);
+	return set_amxstring(amx, params[2], pHash, 32);
+};
 
-	return set_amxstring(amx, params[2], buffer, 32);
-}
+static cell AMX_NATIVE_CALL amx_hash_string(AMX* amx, cell* params)
+{
+	int strLen;
+	char* pStr = get_amxstring(amx, params[1], 0, strLen);
+	hashType Type = (hashType)params[2];
+
+	const char* pHash = hashString((const char *)&pStr[0], strLen, Type);
+	if (!pHash) /** Scripter's fault. Let the server still run. */
+	{
+		LogError(amx, AMX_ERR_NATIVE, "Unable to hash string (\"%s\"). " \
+			"Perhaps \"Type\" parameter isn't valid.", pStr);
+
+		return 0;
+	};
+
+	return set_amxstring(amx, params[3], pHash, params[4]);
+};
+
+static cell AMX_NATIVE_CALL amx_hash_file(AMX* amx, cell* params)
+{
+	int fileNameLen;
+	char* pFileName = get_amxstring(amx, params[1], 0, fileNameLen);
+
+	char fileLocation[256];
+	build_pathname_r(fileLocation, sizeof(fileLocation) - 1, "%s", pFileName);
+
+	hashType Type = (hashType)params[2];
+
+	const char* pHash = hashFile((const char *)&fileLocation[0], Type);
+	if (!pHash)
+	{
+		LogError(amx, AMX_ERR_NATIVE, "There was a problem hashing file (\"%s\"). \
```

Comment:
Be consistent, use the same message as the previous native, and yes remove the "perhaps" sentence.